src/System.Runtime.InteropServices.RuntimeInformation/src/OSPlatform.cs

```diff
@@ -0,0 +1,90 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+namespace System.Runtime.InteropServices
+{
+    public struct OSPlatform : IEquatable<OSPlatform>
+    {
+        private readonly string _osPlatform;
+
+        private const string WindowsName = "WINDOWS";
+        private const string LinuxName = "LINUX";
+        private const string OSXName = "OSX";
+
+        private static readonly OSPlatform s_windows = new OSPlatform(WindowsName);
+        private static readonly OSPlatform s_linux = new OSPlatform(LinuxName);
+        private static readonly OSPlatform s_osx = new OSPlatform(OSXName);
+
+        public static OSPlatform Windows
+        {
+            get
+            {
+                return s_windows;
+            }
+        }
+
+        public static OSPlatform Linux
+        {
+            get
+            {
+                return s_linux;
+            }
+        }
+
+        public static OSPlatform OSX
+        {
+            get
+            {
+                return s_osx;
+            }
+        }
+
+        private OSPlatform(string osPlatform)
+        {
+            if (osPlatform == null) throw new ArgumentNullException("name");
+            if (osPlatform.Length == 0) throw new ArgumentException(SR.Argument_EmptyValue, "name");
```

Comment:
Fixed in https://github.com/dotnet/corefx/commit/2f31aa93f5899790e972b8c571471743337eccc0